api/src/main/java/org/openmrs/obs/handler/ImageHandler.java

```diff
@@ -81,24 +82,31 @@ public Obs getObs(Obs obs, String view) {
 			
 			ComplexData complexData = new ComplexData(file.getName(), img);
 			
-			// Image MIME type
-			try {
-				FileImageInputStream imgStream = new FileImageInputStream(file);
-				Iterator<ImageReader> imgReader = ImageIO.getImageReaders(imgStream);
-				imgStream.close();
-				if (imgReader.hasNext()) {
-					complexData.setMimeType("image/" + imgReader.next().getFormatName().toLowerCase());
-				} else {
-					log.warn("MIME type of " + file.getAbsolutePath() + " is not known");
+			// Get the Mime Type and set it
+			String mimeType = OpenmrsUtil.getFileMimeType(file);
+			
+			// Image MIME type if type couldn't be found		
+			if(mimeType.equals("application/octet-stream")) {
+				try {
```

Comment:
Do we have to wrap this in an if statement? Why not just leave it as before, without the conditional if statement?

Reply:
It depends on what you want. I thought all mimetypes would now be uniformly determined by OpenmrsUtil.getFileMimeType(file) and I only fall back on the old code if that does not determine anything better than application/octet-stream. Do you want me to change it?

By the way: I found a problem in the current OpenMRS core, where AbstractHandler.getOutputFileToWrite() creates weird file names when the file has no extension. For example, the file "Test" becomes "Test.Test" when saved as a complex obs. I will correct this as well and push it in a few minutes.

Reply:
I would be as surgical as possible, just in case. 😊 

As for the found problem, nice catch 👍